torch/distributions/distribution.py

```diff
@@ -116,58 +122,58 @@ def support(self) -> Optional[Any]:
         raise NotImplementedError
 
     @property
-    def mean(self):
+    def mean(self) -> torch.Tensor:
         """
         Returns the mean of the distribution.
         """
         raise NotImplementedError
 
     @property
-    def mode(self):
+    def mode(self) -> torch.Tensor:
         """
         Returns the mode of the distribution.
         """
         raise NotImplementedError(f"{self.__class__} does not implement mode")
 
     @property
-    def variance(self):
+    def variance(self) -> torch.Tensor:
         """
         Returns the variance of the distribution.
         """
         raise NotImplementedError
 
     @property
-    def stddev(self):
+    def stddev(self) -> torch.Tensor:
         """
         Returns the standard deviation of the distribution.
         """
         return self.variance.sqrt()
 
-    def sample(self, sample_shape=torch.Size()):
+    def sample(self, sample_shape: torch.Size = torch.Size()) -> torch.Tensor:
         """
         Generates a sample_shape shaped sample or sample_shape shaped batch of
         samples if the distribution parameters are batched.
         """
         with torch.no_grad():
             return self.rsample(sample_shape)
 
-    def rsample(self, sample_shape=torch.Size()):
+    def rsample(self, sample_shape: torch.Size = torch.Size()) -> torch.Tensor:
```

Comment:
@EPronovost @kit1980 

`torch.Size` is too strict here since something like

```py
dist = torch.distributions.Uniform(0.0, 1.0)
dist.sample((2, 3))
```

is no longer allowed since `(2, 3)` is not a `torch.Size`. This forces users to do 

```py
dist.sample(torch.Size((2, 3)))
```

which is likely not intended, is it? We should use `torch.types._size`

https://github.com/pytorch/pytorch/blob/166b5d3e7c5c230c455dcbcc05c84dd6bc03721b/torch/types.py#L22

as annotation here and in all other occurrences of this annotation.

Reply:
@pmeier Thanks! Do you want to send a PR?